### ISSUE TITLE
plugin buffer monitoring

### ISSUE THREAD

**Issue by rstad** [myeisha/sURL#5]
I may be getting the terminology wrong, or perhaps just talking into the void since this doesn't seem to have been changed for years now, but...

I was trying this out in weechat and found that it seems to behave exactly as expected for IRC channels but doesn't seem to be noticing URLs when they're in a script/plugin buffer (specifically I am using https://github.com/wee-slack/wee-slack)

So for `buffer_plugin buffer_name` of `[irc/freenode] #wee-slack` or `[irc/bitlbee-local] #salt-general` the shortening happens normally but in `[python] redacted.slack.com.#general` I'm not getting anything.  

I'm not sure if this is because of the different naming or something to do with IRC vs not IRC or what. Is this something that I would need to figure out in surl or in the other script?

**Comment by eBrnd** [comment]
No void here, I'll look into it tomorrow.

Quick suspicion: Look at line 109 - it expects the `channel` to start with an `#`, otherwise it thinks it's a message in a query. Maybe because the slack buffer is named `something.slack.com.#channel` this check needs to be changed.

**Comment by eBrnd** [comment]
Ok - I tried surl with wee-slack, and played around a little.

The test at line 109 is not the problem, we don't even get that far. wee-slack doesn't send a `PRIVMSG`. In fact, it doesn't seem to send a signal at all. I tried hooking into `"*"` and got nothing. Normal IRC events, yes, but nothing from wee-slack. The order in which you load plugins doesn't matter either.

I think wee-slack uses some other mechanism to get the text into the buffer, seeing it can even change messages that are already in the buffer if someone edits them through the slack web interface.

I don't know if weechat allows one plugin to send signals to another one, or directly calling functions from one plugin from another one, but I think unless wee-slack adds a function that notifies surl (or any other plugin, we're out of luck.
